### ISSUE TITLE
 [...] changed their display name to  [...]

### ISSUE THREAD

**Issue by witchent** [mautrix/signal#235]
I set ``contact_list_names: allow`` in the config file, but now I get a lot of  [...] changed their display name to  [...]. 
My puppets are changing their names from the contact names to their signal names and back quite often, which is spamming the rooms (and synapse database of course). I did not yet find out when exactly this happens, as I cannot trigger it with a bridge restart nor with a sync command. Any idea how this could be prevented (except for probably disallowing contact list names)?
I am the sole user of the mautrix-signal bridge on my own homeserver.

**Comment by sumnerevans** [comment]
What version of signald are you using? We've made a few changes lately to try and fix that.

**Comment by witchent** [comment]
I am using the latest docker image (signald/signald), so at least for me the fixes do not seem to help.
Can I do something to help debug this?

**Comment by giteichi** [comment]
After updating signald to version 0.17.0-63-b7ea004d and mautrix-signal to 0.2.3 I am having the them problem.

`contact_list_names` is set to prefer

**Comment by helkaluin** [comment]
Is there anything we could do to help debug this? It's been quite a while now and the problem still exists. :/

**Comment by sumnerevans** [comment]
You can try the latest signald and run mautrix-signal with this PR: https://github.com/mautrix/signal/pull/276

We made a few changes to signald that may help things, and the mautrix-signal change adds some more verbose logs that can help debug.

**Comment by giteichi** [comment]
Hi guys,
why is this closed? The problem still exists. There is only a new way for debugging.

**Comment by sumnerevans** [comment]
Well, as of https://github.com/mautrix/signal/commit/b793f24de2af726f00e5599000473ca4df1a4ece it no longer changes to a worse-quality name anymore. So you won't have it changing from a name to a phone number for example.

Additionally, as of [signald@`52a5bae`](https://gitlab.com/signald/signald/-/commit/52a5bae752f79091722410e6dd11ab84e7066364), signald should also provide correct names to mautrix-signal.

If you are running with `master`/`main` on both projects and still experience display names switching back and forth, then let us know.

**Comment by p-rintz** [comment]
I am not running main mautrix-signald, but signald is at commit [7d9179fd](https://gitlab.com/signald/signald/-/commit/7d9179fd4445b303cd2f595dc62bbb61652de2b4)/main.

This issue is still happening on my matrix server. I expect once the next mautrix-signal release comes around, this should be fixed with the storage of the highest quality information. But thought It might be interesting for you @sumnerevans that [52a5bae](https://gitlab.com/signald/signald/-/commit/52a5bae752f79091722410e6dd11ab84e7066364) doesnt appear to fix the issue by itself.

**Comment by maltee1** [comment]
I recently started having a displayname changing back and forth. The had their signal account on my bridge and switched back to using the app. Now the bridge is changing back between whats appears to be their old user name and their new user name. So far I've only been able to get what is probably their new user name from signald.

**Comment by shukon** [comment]
@maltee1 could you explain where to start on debugging fixing this? i'm experiencing this with numbers, that were never themselves registered via my bridge. so unregistering seems not possible or am i misinterpreting the definition?

**Comment by shukon** [comment]
Thanks. I looked the contact up in the signald-database. It's correct in there, but it seems to be fluctuating between names from the contact-lists of two users and the original signal-username that they gave themselves. This is despite setting `contact_list_names` to `disallow`. I guess I'll open up a bug report.

**Comment by maltee1** [comment]
The contact probaby is in the database more than once, but for different signald users. Different signald users would not see the name change back and forth but the bridge syncing from every user makes it change back and forth in matrix.
Check the database for uuid or phone number to find the contact for all users.

**Comment by shukon** [comment]
Ok, so thanks a bundle for your help and sorry for the discussion in a closed issue, will stop commenting in here now, only reporting back the fix.
I managed to solve my problem by overwriting all `given_name`s in the `profiles`-table of `signald.db` where the `recipient`-id was the same as the `rowid` of the corresponding `e164` (phone number) in the `recipients`-table. Let's hope, signald respects those :sweat_smile: 
For other's struggling, assuming 161234567 is the phone-number and anne is the new signal user profile name:
<details>
<summary> show all relevant rows </summary>

```
SELECT profiles.*
FROM profiles
INNER JOIN recipients
ON profiles.recipient = recipients.rowid
WHERE recipients.e164 LIKE '%161234567%';
```

</details>
<details>
<summary> update all relevant rows </summary>

```
UPDATE profiles
SET given_name = 'anne'
FROM profiles AS p
INNER JOIN recipients as r
ON p.recipient = r.rowid
WHERE r.e164 LIKE '%161234567%';
```

</details>

**Comment by maltee1** [comment]
I've only done this once, so I have no idea whether this will work every time.
I needed two attempts. In the first attempt, I changed the profile name to the wrong name and signald changed it back. I then changed the profile name to the correct name and the renaming was gone.
It's entirely possible that it isn't reproducible this way, just maybe check whether you've actually changed the name to the correct name
